files/admin.py

```
@@ -175,7 +182,42 @@ class TopMessageAdmin(admin.ModelAdmin):
+
+        1. First validates that only allowed tags and internal links are present
+        2. Then sanitizes by removing dangerous attributes and cleaning the HTML
+        3. Returns the cleaned value that will be persisted to the database
+        """
+        content = self.cleaned_data["text"]
+        # Validate - will raise ValidationError if disallowed content found
+        validate_internal_html(content)
+        # Sanitize and return cleaned value for persistence
+        return sanitize_html(content)
+
+    class Meta:
```

Ruff (0.13.3)

Redefinition of unused `Meta` from line 196

(F811)